studio/components/interfaces/Auth/RateLimits/RateLimits.tsx

```diff
@@ -0,0 +1,336 @@
+    RATE_LIMIT_SMS_SENT: z.coerce.number().min(0, 'Must be not be lower than 0'),
+  })
+
+  const form = useForm<z.infer<typeof FormSchema>>({
+    resolver: zodResolver(FormSchema),
+    defaultValues: {
+      RATE_LIMIT_TOKEN_REFRESH: 0,
+      RATE_LIMIT_VERIFY: 0,
+      RATE_LIMIT_EMAIL_SENT: 0,
+      RATE_LIMIT_SMS_SENT: 0,
+    },
+  })
+
+  const onSubmit = (data: z.infer<typeof FormSchema>) => {
+    if (!projectRef) return console.error('Project ref is required')
+
+    // [Joshen] Temporarily removing RATE_LIMIT_TOKEN_REFRESH and RATE_LIMIT_VERIFY until BE supports it
+    const payload: Partial<z.infer<typeof FormSchema>> = {}
+    if (data.RATE_LIMIT_EMAIL_SENT !== authConfig?.RATE_LIMIT_EMAIL_SENT) {
+      payload.RATE_LIMIT_EMAIL_SENT = data.RATE_LIMIT_EMAIL_SENT
+    }
+    if (data.RATE_LIMIT_SMS_SENT !== authConfig?.RATE_LIMIT_SMS_SENT) {
+      payload.RATE_LIMIT_SMS_SENT = data.RATE_LIMIT_SMS_SENT
+    }
+    updateAuthConfig({ projectRef, config: payload }, { onSuccess: () => form.reset(data) })
+  }
+
+  useEffect(() => {
+    if (isSuccess) {
+      form.reset({
+        RATE_LIMIT_TOKEN_REFRESH: 30,
+        RATE_LIMIT_VERIFY: 30,
+        RATE_LIMIT_EMAIL_SENT: authConfig.RATE_LIMIT_EMAIL_SENT,
+        RATE_LIMIT_SMS_SENT: authConfig.RATE_LIMIT_SMS_SENT,
+      })
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isSuccess])
+
+  return (
+    <div>
+      <FormHeader
+        title="Rate Limits"
+        description="Safeguard against bursts of incoming traffic to prevent abuse and maximize stability"
+        docsUrl="https://supabase.com/docs/guides/platform/going-into-prod#rate-limiting-resource-allocation--abuse-prevention"
+      />
+
+      {isError && <AlertError subject="Failed to retrieve auth config rate limits" error={error} />}
+
+      {isLoading && <GenericSkeletonLoader />}
+
+      {isSuccess && (
+        <Form_Shadcn_ {...form}>
+          <form id={formId} className="space-y-4" onSubmit={form.handleSubmit(onSubmit)}>
+            <FormPanel
+              footer={
+                <div className="flex py-4 px-8">
+                  <FormActions
+                    form={formId}
+                    isSubmitting={isUpdatingConfig}
+                    hasChanges={form.formState.isDirty}
+                    handleReset={() => form.reset()}
+                    disabled={!canUpdateConfig}
+                    helper={
+                      !canUpdateConfig
+                        ? 'You need additional permissions to update authentication settings'
+                        : undefined
+                    }
+                  />
+                </div>
+              }
+            >
+              <FormSection
+                id="email-sent"
+                header={
+                  <FormSectionLabel
+                    description={
+                      <p className="text-foreground-light text-sm">
+                        How many emails can be sent per hour
```

Comment:
I've always wondered if this is a global limit or per user? Might be worth updating this message accordingly

Reply:
gotcha! updated the descriptions  👍